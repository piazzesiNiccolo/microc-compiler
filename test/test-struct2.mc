struct complex{
    float real;
    float img;
};

struct complex add(struct complex n1, struct complex n2) {
    struct complex temp;
    temp.real = n1.real + n2.real;
    temp.img = n1.img + n2.img;
    return temp;
}

int main(){
    struct complex n1;
    struct complex n2;
    n1.real = 2.1;
    n1.img = 1.3;

    n2.real = .5;
    n2.img = .9;
    struct complex r = add(n1, n2);
    printfloat(r.real);
    printfloat(r.img);


}