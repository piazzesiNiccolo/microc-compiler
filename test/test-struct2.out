2.600000
2.200000
